File: WizQuixTwitchPaintApiService/Logic/Commands/GetCanvasCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WizQuixTwitchPaintApiService.Logic.Commands
{
    /*
        command: getcanvas
        on success: info getcanvas <canvas:string> // canvas in format: <<coordinate:string>=<colorname:string>;[]> coordinate in formal <letter:char><number:int>
    */
    public class GetCanvasCommand : ACommand
    {
        public GetCanvasCommand(WebClient client) : base("getcanvas", client) { }

        public override async Task Execute(string[] args)
        {
            string canvas = Client.JoinedRoom.GetCanvasAsString();
            await Client.SendMessage($"info getcanvas {canvas}");
        }
    }
}

File: WizQuixTwitchPaintExtension/Classes/Models/Commands/GetCanvasCommand.ts
class GetCanvasCommand extends ACommand
{
    public constructor(client:WebClient)
    {
        super("getcanvas", client);
    }

    public OnInfo(data: string[])
    {
        let canvas = data.join(" ");
        this.Client.Canvas.ParceCanvas(canvas);
    }

    public OnError(code: number, error: string)
    {
        
    }
    
}